src/poll.rs

```diff
@@ -23,26 +34,82 @@ impl PollFd {
 
+    /// Returns the events that occured in the last call to `poll` or `ppoll`.
     pub fn revents(&self) -> Option<EventFlags> {
         EventFlags::from_bits(self.pollfd.revents)
     }
 }
 
 libc_bitflags! {
+    /// These flags define the different events that can be monitored by `poll` and `ppoll`
     pub flags EventFlags: libc::c_short {
+        /// There is data to read.
         POLLIN,
+        /// There is some exceptional condition on the file descriptor.
+        ///
+        /// Possibilities include:
+        ///
+        /// *  There is out-of-band data on a TCP socket (see
+        ///    [tcp(7)](http://man7.org/linux/man-pages/man7/tcp.7.html)).
+        /// *  A pseudoterminal master in packet mode has seen a state
+        ///    change on the slave (see
+        ///    [ioctl_tty(2)](http://man7.org/linux/man-pages/man2/ioctl_tty.2.html)).
+        /// *  A cgroup.events file has been modified (see
+        ///    [cgroups(7)](http://man7.org/linux/man-pages/man7/cgroups.7.html)).
         POLLPRI,
+        /// Writing is now possible, though a write larger that the
+        /// available space in a socket or pipe will still block (unless
+        /// `O_NONBLOCK` is set).
         POLLOUT,
+        /// Equivalent to [`POLLIN`](constant.POLLIN.html)
         POLLRDNORM,
+        /// Equivalent to [`POLLOUT`](constant.POLLOUT.html)
         POLLWRNORM,
+        /// Priority band data can be read (generally unused on Linux).
         POLLRDBAND,
+        /// Priority data may be written.
         POLLWRBAND,
+        /// Error condition (only returned in
+        /// [`PollFd::revents`](struct.PollFd.html#method.revents);
+        /// ignored in [`PollFd::new`](struct.PollFd.html#method.new)).
+        /// This bit is also set for a file descriptor referring to the
+        /// write end of a pipe when the read end has been closed.
         POLLERR,
+        /// Hang up (only returned in [`PollFd::revents`](struct.PollFd.html#method.revents);
+        /// ignored in [`PollFd::new`](struct.PollFd.html#method.new)).
+        /// Note that when reading from a channel such as a pipe or a stream
+        /// socket, this event merely indicates that the peer closed its
+        /// end of the channel.  Subsequent reads from the channel will
+        /// return 0 (end of file) only after all outstanding data in the
+        /// channel has been consumed.
         POLLHUP,
+        /// Invalid request: `fd` not open (only returned in
+        /// [`PollFd::revents`](struct.PollFd.html#method.revents);
+        /// ignored in [`PollFd::new`](struct.PollFd.html#method.new)).
         POLLNVAL,
     }
 }
 
+/// `poll` waits for one of a set of file descriptors to become ready to perform I/O.
+/// ([`poll(2)`](http://man7.org/linux/man-pages/man2/poll.2.html))
+///
+/// `fds` is a slice containing all [`PollFd`](struct.PollFd.html) to poll.
+/// The function will return as soon as one event registered in one of the `PollFd` occurs.
```

Comment:
"The function will return as soon as any events occur for any of the `PollFd`s." Something like that, this is a bit more readable I think.